mlrun/api/db/sqldb/db.py

```diff
@@ -2122,6 +2122,36 @@ def _latest_uid_filter(self, session, query):
             ),
         )
 
+    def _add_artifact_name_and_iter_query(self, query, name=None, iter=None):
+        if not name and not iter:
+            return query
+
+        if name and name.startswith("~"):
+            # Like query
+            iter_prefix = f"{iter}-" if iter else ""
+            return query.filter(Artifact.key.ilike(f"{iter_prefix}%{name[1:]}%"))
+
+        # From here on, it's either exact name match or no name
+        if iter:
+            if name:
+                return query.filter(Artifact.key == f"{iter}-{name}")
+            return query.filter(Artifact.key.ilike(f"{iter}-%"))
+
+        # Exact match, no iter specified
+        # Escape special chars (_,%) since we still need to do a like query because of the iter.
+        # Also limit length to len(str) + 4, assuming iter is < 100 - this helps filter the situations where
```

Comment:
Why 4 ?
2 for digits (representing a number less than 100), 1 for the hyphen, and ? (add the explanation in the comment)

Reply:
I actually do 
`func.length(Artifact.key) < len(name) + 4`
So, since it's < rather than <= then it's two digits and the hyphen. The comment may be a bit misleading if not reading the code, I guess - I will change it if it's that critical.